[Monitor.Query] Update Preview Swagger File

nisha-bhatia opened the pull request:
# Data Plane API - Pull Request
<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

nisha-bhatia committed:
update swagger

nisha-bhatia commented on specification/monitor/data-plane/Microsoft.Insights/preview/2023-11-01-preview/metricBatch.json:
```diff
@@ -0,0 +1,562 @@
+      ],
+      "description": "The result data of a query."
+    },
+    "QueryBatchTimeSeriesElement": {
```
This is a duplicate of the pre-existing type TimeSeriesElement except "values" is now called "data" if we revert to "values" we can remove this duplicate

gracewilcox committed:
batch update

gracewilcox committed:
update old swagger

gracewilcox commented on specification/monitor/resource-manager/Microsoft.Insights/preview/2023-11-01-preview/metricsDefinitions_API.json:
```diff
@@ -0,0 +1,273 @@
+    "swagger": "2.0",
+    "info": {
+      "title": "MonitorManagementClient",
+      "version": "2018-01-01"
```
We need to decide the version for these swaggers. Can we use the old version or must we update to the new one?

gracewilcox replied:
decided on the version offline with Srikanta

nisha-bhatia committed:
Update metricBatch.json

nisha-bhatia committed:
Update metricBatch.json

nisha-bhatia committed:
Update metrics_API.json

nisha-bhatia committed:
wip

nisha-bhatia committed:
Update metricBatch.json

nisha-bhatia committed:
wip

nisha-bhatia committed:
update path

nisha-bhatia committed:
Update metricBatch.json

nisha-bhatia committed:
Update metricBatch.json

nisha-bhatia committed:
Update metricBatch.json

nisha-bhatia committed:
Update commonDefinitions.json

gracewilcox commented on specification/monitor/commonDefinitions.json:
```diff
@@ -0,0 +1,178 @@
+          "type": "string",
+          "description": "Error message encountered querying this specific metric."
+        },
+        "unit": {
```
"Breaking change" for the metricsBatch.json. Currently, `unit` has modelAsString set to false. We could have two separate `Unit` models, but I think it's better to make the change and set it as true.

ToddKingMSFT committed:
Change metricBatch.json endpoint string format from "uri" to "url"

Change metricBatch.json endpoint string format from "uri" to "url"

ToddKingMSFT committed:
Remove duration format from interval parameter and properties

Remove duration format from interval parameter and properties. These were improperly decorated as format: "duration" however now that model validation is actually running against parameters our model validation is failing as "FULL" has been a long time accepted string to this parameter and has a lot of usage.

mikekistler commented:
Removing the new-api-version label as this PR does not create a new API version.

ToddKingMSFT commented:
Hello @tjprescott can you review this and give us ARM signoff?
For context these are changes requested by the SDK team to assist in creating a new version of the metrics SDKS for both the ARM and dataplane metrics APIs. There were no API changes as part of this, it's all formatting changes for the SDKs.

rkmanda reviewed (approved):
:shipit:
